server/channels/store/sqlstore/job_store.go

```diff
@@ -201,43 +202,29 @@ func (jss SqlJobStore) UpdateStatusOptimistically(id string, currentStatus strin
 	return true, nil
 }
 
-func (jss SqlJobStore) Get(id string) (*model.Job, error) {
+func (jss SqlJobStore) Get(c *request.Context, id string) (*model.Job, error) {
```

Comment:
WDYT passing only the logger here, ideally store layer should be decoupled from `request` package.

PS: There are several occurrences like this, my comment applies to others as well.

Reply:
> ideally store layer should be decoupled from `request` package.

Can you elaborate on why the `request.Context` should not get passed to the `store` layer? We surely want the `logger` why not also pass down the rest?

Reply:
There is also a proposal on Community (which I can't link currently can't link :wink:) to add `request.Context` to the whole stack.

Reply:
Hmm, my idea was originating from basic idea of separating the store layer from upper layers. `request.Context` is more likely about application and user. I'd like to avoid creating the dependency as much as I can. In this case we just want to pass logger, we actually don't require the context itself.

Reply:
There are two things that are needed in the store layer: a `mlog.Logger` and a `context.Context` for cancellation.
 While we could add these two things as arguments, it seems convenient to pass them as a `request.Context`. Also https://community.mattermost.com/core/pl/sw1pjdus9jrmfqk81g9yqstzie is the conversation I was looking for.

Reply:
Thanks for pointing out the discussion. I wasn't thinking passing the context for cancellation, but if it's the case then it makes sense.